vsphere/structure_helper.go

```diff
@@ -0,0 +1,38 @@
+package vsphere
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+// sliceInterfacesToStrings converts an interface slice to a string slice. The
+// function does not attempt to do any sanity checking and will panic if one of
+// the items in the slice is not a string.
+func sliceInterfacesToStrings(s []interface{}) []string {
+	var d []string
+	for _, v := range s {
+		d = append(d, v.(string))
+	}
+	return d
+}
+
+// sliceStringsToInterfaces converts a string slice to an interface slice.
+func sliceStringsToInterfaces(s []string) []interface{} {
+	var d []interface{}
+	for _, v := range s {
+		d = append(d, v)
+	}
+	return d
+}
+
+// mergeSchema merges the map[string]*schema.Schema from src into dst. Safety
+// against conflicts is enforced by panicing.
+func mergeSchema(dst, src map[string]*schema.Schema) {
```

Comment:
_minor_ longer term this probably belongs in TF Core with some tests, but it's not a blocker by any means

Reply:
I think this might be another job for the [schema transformer](https://github.com/terraform-providers/terraform-provider-vsphere/pull/138#discussion_r136401357)! :wink: